Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -365,39 +372,148 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
-    # If we have a file - Try to use it.
-    # If there's no file, or we cannot use it - Find a free port
-    browser = None
-    chrome_port = read_info_file(PORT_FILE_PATH)
-    if chrome_port:
-        browser = is_chrome_running_locally(chrome_port)
-        if browser:
-            return browser, chrome_port
+def chrome_manager():
+    instance_id = demisto.callingContext.get('context', {}).get('IntegrationInstanceID')
+    chrome_options = demisto.callingContext.get('params', {}).get('chrome_options')
+    demisto.log(f"in chrome_manager {chrome_options=}, {instance_id=}")  # TODO: Remove log
+    demisto.log(f"{instance_id=}")  # TODO: Remove log
+
+    if not instance_id:
+        instance_id = 'NO-INSTANCE-ID'
+    if not chrome_options:
+        chrome_options = 'None'
+
+    info = read_info_file(CHROME_INSTANCES_FILE_PATH)
+    demisto.log(f"{info=}")  # TODO: Remove log
+
+    instance_id_to_chromes_options, chrome_options_to_port, instance_id_to_port = get_chrome_instances_info_dictionaries(info)
+
+    if info:
+        demisto.log("there is an info")  # TODO: Remove log
+        info_without_new_lines = info.strip()
+        splitted_info = info_without_new_lines.splitlines()
+        _, instances_id_tuple, chromes_options_tuple = zip(*[line.strip().split('\t') for line in splitted_info])
+        instances_id = list(instances_id_tuple) if instances_id_tuple else []
+        chromes_options = list(chromes_options_tuple) if chromes_options_tuple else []
+
+        if not chromes_options and instances_id:
+            chromes_options.append('None')
+
+        demisto.log(f"{chromes_options=}, {instances_id=}")  # TODO: Remove log
+
+    if not info or (chrome_options in chromes_options and instance_id not in instances_id) or (
+        chrome_options not in chromes_options and instance_id not in instances_id):
+        """
+        When:
+            case 1: file is empty, that means no chrome open on the machine
+            or
+            case 2: new instance with the same existing chrome configuration as another instance
+            or
+            case 3: chrome configuration does not exist and integration instance does not exist in the file
+        Then:
+            create new browser
+        """
+
+        # TODO: remove the block below
+        demisto.log("in chrome manager first if - case 1,2,3")
+        if not info:
+            demisto.log("first if - no info file found (case 1)")
         else:
-            write_info_file(PORT_FILE_PATH, '')
+            demisto.log("first if - info file exist")
+            if chrome_options in chromes_options and instance_id not in instances_id:
+                demisto.log(f"chrome_options in chromes_options and instance_id not in instances_id (case 2)")
+            elif chrome_options not in chromes_options and instance_id not in instances_id:
+                demisto.log(f"chrome_options not in chromes_options and instance_id not in instances_id (case 3)")
```

Comment:
Missing logic?

Reply:
This block is purely for log's for myself so I can differ between the cases. If it is acceptable, I will change the log's there to demisto.info. And if it is not acceptable to write an if/else block without logic, I will omit this part when the pr is confirmed.

Reply:
If these log messages can be meaningful in a future bug - Then leave them here.